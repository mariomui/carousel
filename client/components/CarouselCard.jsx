import React from 'react';
import styled from 'styled-components';


const Image = styled.img`
  border: 1px solid black;
  margin: 2px;
  :nth-of-type(n+2) {
    width: 21%;
    height: 21%;
  }
  :first-of-type {
    float: left;
    padding: 0px
  }
`;
const Wrapper = styled.div`
  display: grid;
`;
const CarouselCard = ({ imageUrl, key }) => (
  // <Wrapper className="picture-row">
  <Image className="media-element" src={imageUrl} key={key} alt="pictures" />
  // </Wrapper>
);

export default CarouselCard;
