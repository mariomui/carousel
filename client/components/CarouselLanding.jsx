/* eslint-disable no-useless-constructor */
/* eslint-disable class-methods-use-this */
import React from 'react';

class Landing extends React.Component {
  constructor() {
    super();
    this.state = {
      image_url: [],
    };
  }

  render() {
    return (
      <div> Hot module repalcement!! </div>
    );
  }
}

export default Landing;
