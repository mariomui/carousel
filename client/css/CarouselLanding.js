import styled from 'styled-components';

module.exports = {
  Grid: styled.div`
    .try {
      color: green;
    };
  `,
};
