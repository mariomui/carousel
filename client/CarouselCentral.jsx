import React, { Component } from 'react';
import { BrowserRouter as Brouter, Switch, Route } from 'react-router-dom';
import styled, { createGlobalStyle } from 'styled-components';
import CarouselLanding from './components/CarouselLanding';
import getPicturesFromServer from './components/Temp';
import CarouselCard from './components/CarouselCard';
import Modal from './components/Modal'

// eslint-disable-next-line react/prefer-stateless-function
class CarouselCentral extends Component {
  // Brouters only allow one child element
  constructor() {
    super();
    this.state = {
      modal: false,
      imageUrls: [],
    };
    this.handleModalClick = this.handleModalClick.bind(this);
    this.handleGetPictures = this.handleGetPictures.bind(this);
  }

  handleModalClick(e) {
    e.preventDefault();
    const { modal } = this.state;
    this.setState({
      modal: !modal,
    });
  }

  handleGetPictures(imageUrls) {
    this.setState({
      imageUrls: imageUrls,
    });
  }

  render() {
    const { modal, imageUrls } = this.state;

    return (
      <div>
        <CarouselLanding handleGetPictures={(imageUrls) => { this.handleGetPictures(imageUrls); }} handleModalClick={this.handleModalClick} />
        {modal && <Modal imageUrls={imageUrls} handleModalClick={this.handleModalClick} />}
      </div>
    );
  }
}


function CarouselLauncher() {
  return (
    <Brouter>
      <Route component={CarouselCentral} />
    </Brouter>
  );
}

export default CarouselLauncher;
