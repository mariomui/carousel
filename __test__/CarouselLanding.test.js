import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { expect } from 'chai';
import Adapter from 'enzyme-adapter-react-16';
// import renderer from 'react-test-renderer';
import { Z_TREES } from 'zlib';
import CarouselCentral from '../client/CarouselCentral';
import CarouselLanding from '../client/components/CarouselLanding';
import CarouselCard from '../client/components/CarouselCard';

Enzyme.configure({ adapter: new Adapter() });

// describe('hey', () => {

// });
describe('Carousel Landing', () => {
  // const tree = mount(<CarouselLanding />);

  it('to render five carousel card components', () => {
    const tree = shallow(<CarouselLanding />);
    tree.setState({
      imageUrls: [
        'http://google.com',
        'http://google.com',
        'http://google.com',
        'http://google.com',
      ],
    });
    expect(tree.find(CarouselCard)).to.have.lengthOf(4);
  });
});

// test('This tests how many nodes are on the screen', () => {
//   const tree = shallow(<CarouselLanding />);
// });
