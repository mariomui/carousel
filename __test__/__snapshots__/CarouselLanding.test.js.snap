// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Landing to exist 1`] = `<div />`;
